### PR TITLE
[match][hotfix] remove the `renew_expired_certs` option introduced in #21691 and revert the default behavior while we address issues with it

### PR DESCRIPTION
**Pull request by nekrich**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Removes `renew_expired_certs` #21691, but behavior stays the same.
Doing this in favor of #21809, that doesn't require this option any more

### CODE REVIEW

File: match/lib/match/runner.rb [hunk 1]
```diff
@@ -84,12 +84,12 @@ def run(params)
       end
 
       # Certificate
-      cert_id = fetch_certificate(params: params, renew_expired_certs: params[:renew_expired_certs])
+      cert_id = fetch_certificate(params: params, renew_expired_certs: true)
```
Comment (rogerluan) [inline match/lib/match/runner.rb:87]: If we wanted to keep the previous behavior, we'd pass `false` here, no?

File: match/lib/match/runner.rb [hunk 2]
```diff
@@ -84,12 +84,12 @@ def run(params)
...
 
       # Mac Installer Distribution Certificate
       additional_cert_types = params[:additional_cert_types] || []
       cert_ids = additional_cert_types.map do |additional_cert_type|
-        fetch_certificate(params: params, renew_expired_certs: params[:renew_expired_certs], specific_cert_type: additional_cert_type)
+        fetch_certificate(params: params, renew_expired_certs: true, specific_cert_type: additional_cert_type)
```
Comment (rogerluan) [inline match/lib/match/runner.rb:92]: Same?

**Review by rogerluan — CHANGES_REQUESTED** [review]
Thanks for opening this PR, we need to carefully review whether we want to revert the feature (aka merge this PR) or merge your other one (that fixes it) 😬 

Can you review my question below? 🙏

**Comment by nekrich** [comment]
@rogerluan , sorry for the delay. Changed all to false and removed auto-renewal test.

File: match/lib/match/runner.rb [hunk 3]
```diff
@@ -172,7 +172,7 @@ def fetch_certificate(params: nil, renew_expired_certs: false, specific_cert_typ
       is_cert_renewable = is_authenticated_with_login || is_cert_renewable_via_api
 
       # Validate existing certificate first.
-      if renew_expired_certs && is_cert_renewable && storage_has_certs
+      if renew_expired_certs && is_cert_renewable && storage_has_certs && !params[:readonly]
```
Comment (rogerluan) [inline match/lib/match/runner.rb:175]: Can you remind me why we need this change? 👀  @nekrich

**Review by rogerluan — APPROVED** [review]
LGTM, just left a question. @lacostej I think this is good to go in case we need to deploy 😬 do you think we should merge it in and then revert it + fix the error in the other related PR?

File: match/lib/match/runner.rb (hunk 3 above)
Comment (lacostej) [reply match/lib/match/runner.rb:175]: @rogerluan well, we shouldn't be deleting things when we are in readonly mode :)

**Review by lacostej — APPROVED** [review]

**Pull request merged**